### DISCUSSION TITLE
ch11-Logging and Monitoring STARTUP_MODE = "OnPremises" Run after Sql Container login error

### DISCUSSION THREAD

**Discussion by kssoftwangwg** [PacktPublishing/Pragmatic-Microservices-with-CSharp-and-Azure#214] [Q&A] [answered]
Hi, can you please help me solve the problem I'm having?

Select "Codebreaker.AppHost:OnPremises" setting after, run it, Codebreaker.GameAPIs[0] has error:
2024-09-27T14:21:50.6530000 2024-09-27T22:21:50.6361770 fail: Codebreaker.GameAPIs[0]
2024-09-27T14:21:50.6530000       Error updating database
2024-09-27T14:21:50.6530000       Microsoft.Data.SqlClient.SqlException (0x80131904): A connection was successfully established with the server, but then an error occurred during the pre-login handshake. (provider: TCP Provider, error: 0 - Undefined error: 0).  
error line at Codebreaker.GameAPIs.ApplicationServices.CreateOrUpdateDatabaseAsync(WebApplication app).

See Sql Container logs on docker:  Login failed for user 'sa'. Reason: Failed to open the explicitly specified database 'CodebreakerSql'. [CLIENT: 192.168.215.1], maybe it's missing ef core Migrations files in project Codebreaker.GameAPIs?
 
If true reason of this, how to generate them?

**Comment by christiannagel** [comment, marked as answer]
@kssoftwangwg The reason looks like the password for the sa account. In case the sa account was configured differently, or random before (this is done without specifying a password).
Delete the volume and start the application again. You can delete the volume using the Docker desktop, or with the command line `docker volume`.

I hope this helps.
I'm moving this to the discussions.
